Полезные в дальнейшем сокращения для определяющих соотношений:

In [ ]:
sim := function(a, b) # sim(a, b) = 1 <=> a ~ b
    return a*b*a*b^-1*a^-1*b^-1;
end;;

square := function(x, y) # square(x, y) = [1, 1, 1, 1] <=> x □ y, т.е. <x, y> = A5
    return [x^3, y^3, (x*y)^5, (x^y*x)^2];
end;;

triangle := function(x, y) # triangle(x, y) = [1, 1, 1, 1] <=> x △ y, т.е. <x, y> = SL(2, 5)
    # работает, только когда имена порождающих переменных "x" и "y", естественно, не всегда это удобно. Возвращает список [ x^3, y^3, (y*x)^2*y*(x^-1*y^-1)^2*x^-1, (y^-1*x)^2*y^-1*(x^-1*y)^2*x^-1 ] :
    # return ParseRelators([x, y], "x^3 = y^3 = 1, xyxyx = yxyxy, xy^-1xy^-1x = y^-1xy^-1xy^-1");
    return [ x^3, y^3, (y*x)^2*y*(x^-1*y^-1)^2*x^-1, (y^-1*x)^2*y^-1*(x^-1*y)^2*x^-1 ]; # работает независимо от меток переменных x и y
end;;

comm := function(x, y)
    return Comm(x, y);
end;;

In [ ]:
# пример использования

F := FreeGroup("x", "y");; #для ParseRelators в triangle(x, y) нужно, чтобы порождающие имели однобуквенные имена "x" и "y"
x := F.1;; y := F.2;;

Print(StructureDescription(F / [x^3, y^3, sim(x, y)]), "\n"); # SL(2,3)
Print(StructureDescription(F / square(x, y)), "\n"); # A5
Print(StructureDescription(F / triangle(x, y)), "\n"); # SL(2, 5)

SL(2,3)
A5
SL(2,5)


Порождающие $SL_2(3)$:

In [1238]:
a := [[Z(3)^0, Z(3)^0], [0*Z(3), Z(3)^0]];;
b := [[Z(3)^0, 0*Z(3)], [-Z(3)^0, Z(3)^0]];;

Group(a, b) = SpecialLinearGroup(2, 3);
sim(a, b) = One(Group(a, b)); #true, то есть a ~ b

true

true

**Лемма 1.**
Пусть $G$ --- $\mathfrak{M}_4$-группа, $x,y,z \in D$, $y\not \in \langle x,z \rangle$, $G=\langle x,y,z \rangle$ и $[x,y]=1$. Тогда
1) Если $\langle x,z \rangle \simeq A_4$, то $G \simeq A_6$.
2) Если $\langle x,z \rangle \simeq A_5$, то $G$ является фактором одной из групп $U,V,A_6,A_7$.
3) Если $\langle x,z \rangle \simeq SL_2(3)$, то

**Доказательство.**

1. Подстановки $x= (1,2,3)$, $y=(4,5,6)$ и $z=(2,3,4)$ удовлетворяют требуемым условиям и порождают $A_6$:

In [ ]:
x := (1, 2, 3);; y := (4, 5, 6);; z := (2, 3, 4);;
G := Group(x, y, z);;
H := Group(x, z);;

H = AlternatingGroup(4);
IsSubset(ConjugacyClass(G, x), [y, z]);
(y in Group(x, z)) = false;
Comm(x, y) = ();
G = AlternatingGroup(6);

true

true

true

true

true

Пусть $K=\{ x,y,z \mid x^3,y^3,z^3,[x,y],(xz)^2 \}$ --- группа, заданная соотношениями из утверждения.

In [ ]:
F := FreeGroup("x", "y", "z");;

x := F.1;; y := F.2;; z := F.3;;

relsK := [x^3, y^3, z^3, comm(x, y), (x*z)^2];;
K := F / relsK;;

Заметим, что $K/\langle (yz)^3 \rangle \simeq A_4 \times A_4$:

In [1259]:
x := F.1;; y := F.2;; z := F.3;;

relsK1 := Concatenation(relsK, [(y*z)^3]);;
K1 := F / relsK1;;

In [ ]:
A4 := AlternatingGroup(4);;
A4xA4 := DirectProduct(A4, A4);;

# iso := IsomorphismGroups(K1, A4xA4);;
# Вывод: [ f1, f2, f3 ] -> [ (1,4,3)(6,8,7), (6,7,8), (1,4,2)(5,7,6) ]

f1 := K1.1;; f2 := K1.2;; f3 := K1.3;;
iso := GroupHomomorphismByImages(K1, A4xA4, [ f1, f2, f3 ], [ (1,4,3)(6,8,7), (6,7,8), (1,4,2)(5,7,6) ]);;

Image(iso) = A4xA4;

true

поэтому, заменяя, если необходимо, $y$ на $y^{-1}$, можно считать, что 
$\langle y,z \rangle \not \in \{3^2,3^{1+2},A_4 \}$.

***\% BD: про замену необязательно, видимо***

Кроме того, $K/\langle (yz)^4 \rangle = \langle yK,zK \rangle \simeq A_4 \times A_6$,

In [ ]:
x := F.1;; y := F.2;; z := F.3;;

relsK2 := Concatenation(relsK, [(y*z)^4]);;
K2 := F / relsK2;;

In [1279]:
A4 := AlternatingGroup(4);;
A6 := AlternatingGroup(6);;
A4xA6 :=  DirectProduct(A4, A6);;

# iso := IsomorphismGroups(K2, A4xA6);;
# Вывод: [ f1*f3, f1*(f2^-1*f3^-1)^2 ] -> [ (1,2)(3,4)(6,7)(8,9), (2,4,3)(5,6,7,8)(9,10) ]

f1 := K2.1;; f2 := K2.2;; f3 := K2.3;;
iso := GroupHomomorphismByImages(K2, A4xA6, [f1*f3, f1*(f2^-1*f3^-1)^2 ], [ (1,2)(3,4)(6,7)(8,9), (2,4,3)(5,6,7,8)(9,10) ]);;

Image(iso) = A4xA6;

true

поэтому $\langle y,z \rangle \not \in \{SL_2(3) \}$.

***\% BD: почему?***

Наконец, $K/\langle yzyzy=zyzyz \rangle \simeq 3 \times A_6$,

In [ ]:
x := F.1;; y := F.2;; z := F.3;;

relsK3 := Concatenation(relsK, ParseRelators([x, y, z], "yzyzy = zyzyz"));;
K3 := F / relsK3;;

In [ ]:
A6 := AlternatingGroup(6);;
C3 := CyclicGroup(3);;
C3xA6 := DirectProduct(C3, A6);;

# iso := IsomorphismGroups(K3, C3xA6);
# Вывод: [ y^-2*x^-1*(x^-2*y*z)^6*x*y^2, x^-1*y*z*x^-1 ] -> [ DirectProductElement( [ <identity> of ..., (2,3)(5,6) ] ),   DirectProductElement( [ f1, (1,2)(3,4,5,6) ] ) ]

f1 := C3.1;;
x := K3.1;; y := K3.2;; z := K3.3;;
iso := GroupHomomorphismByImages(K3, C3xA6, [ y^-2*x^-1*(x^-2*y*z)^6*x*y^2, x^-1*y*z*x^-1 ], [ DirectProductElement( [One(C3), (2,3)(5,6) ] ),   DirectProductElement( [ f1, (1,2)(3,4,5,6) ] ) ]);;

Image(iso) = C3xA6; 

true

при этом $\langle yK,zK \rangle \simeq GL_2(4)$. ***\% BD: здесь $K$ уже с учётом нового соотношения, нужно переименовать***

In [1300]:
x := K3.1;; y := K3.2;; z := K3.3;;
Gyz := Group(y, z);;
GL24 := GeneralLinearGroup(2, 4);;

# iso := IsomorphismGroups(Gyz, GL24);
# [ y^-1*z^-2*y*z*y^2*z*(z*y)^2*z^2*y, y*z*(z*y)^2*y*z ] -> [ [ [ Z(2^2), Z(2)^0 ], [ Z(2^2), Z(2^2) ] ],   [ [ Z(2^2), 0*Z(2) ], [ 0*Z(2), Z(2)^0 ] ] ]

iso := GroupHomomorphismByImages(Gyz, GL24, [ y^-1*z^-2*y*z*y^2*z*(z*y)^2*z^2*y, y*z*(z*y)^2*y*z ], [ [ [ Z(2^2), Z(2)^0 ], [ Z(2^2), Z(2^2) ] ],   [ [ Z(2^2), 0*Z(2) ], [ 0*Z(2), Z(2)^0 ] ] ]);;
Image(iso) = GL24;

true

2. Пусть $\langle x,z \rangle \simeq A_5$.


In [ ]:
A5 := AlternatingGroup(5);;

A5_gens := [];;

for x in A5 do
    if Order(x) <> 3 then continue; fi;

    for z in AsList(ConjugacyClass(A5, x)) do
        if Group(x, z) = A5 then
            Add(A5_gens, [x, z]);;
        fi;
    od;
od;

In [ ]:
# Print(A5_gens);

Тогда $\langle x,x^z \rangle \simeq A_4$:

In [1305]:
LoadPackage("sonata");;

for gens in A5_gens do
    x := gens[1];; z := gens[2];;
    if not IsIsomorphicGroup(Group(x, x^z), AlternatingGroup(4)) then
        Print("Not A4: <", x, " ", x^z, "> = ", StructureDescription(Group(x, x^z)), "\n"); #вывод пустой
    fi;
od;

\% *На самом деле, если подумать, непосредственно понятно: элементы порядка 3 в $A_5$ --- тройные циклы, и строение порождённых ими подгрупп описано во введении. То есть $x$ и $z$ имеют ровно один перемещаемый элемент. $(123)(345)(321) = (245)$, то есть при сопряжении $x$ при помощи $z$ общий перемещаемый элемент заменяется, а два других остаются, и теперь $x$ и $x^z$ порождают $A_4$.*

и по доказанному либо $[x^z,y]=1$; либо $\langle x,x^z,y \rangle \simeq A_6$ и $\langle y,x^z \rangle \simeq A_5$.

 Для $\langle x,x^{z^{-1}} \rangle \simeq A_4$ можно рассуждать аналогично.
Пусть снова $K=\{ x,y,z \mid x^3,y^3,z^3,[x,y],(x^zx)^2,(xz)^5 \}$ --- группа, заданная соотношениями из утверждения.

In [ ]:
x := F.1;; y := F.2;; z := F.3;;

relsK := [x^3,y^3,z^3,comm(x,y),(x^z*x)^2,(x*z)^5];;
K := F / relsK;;

Заметим, что $K/\langle [x^z,y],(yz)^3 \rangle \simeq GL_2(4)$ и $K/\langle [x^z,y],(yz)^4 \rangle \simeq 1$.

In [ ]:
x := F.1;; y := F.2;; z := F.3;;

relsK1 := Concatenation(relsK, [comm(x^z, y), (y*z)^3]);;
K1 := F / relsK1;;

GL24 := GeneralLinearGroup(2, 4);;

# iso := IsomorphismGroups(K1, GL24);
# Вывод: [ x^-1*z^-2*x^-1*z*x^2*z*(z*x)^2*z^2*x, y^2*z ] -> [ [ [ Z(2^2), Z(2)^0 ], [ Z(2^2), Z(2^2) ] ],   [ [ Z(2^2), 0*Z(2) ], [ 0*Z(2), Z(2)^0 ] ] ]

x := K1.1;; y := K1.2;; z := K1.3;;
iso := GroupHomomorphismByImages(K1, GL24, [ x^-1*z^-2*x^-1*z*x^2*z*(z*x)^2*z^2*x, y^2*z ] , [ [ [ Z(2^2), Z(2)^0 ], [ Z(2^2), Z(2^2) ] ],   [ [ Z(2^2), 0*Z(2) ], [ 0*Z(2), Z(2)^0 ] ] ]);;

Image(iso) = GL24;

true

In [1327]:
x := F.1;; y := F.2;; z := F.3;;

relsK2 := Concatenation(relsK, [comm(x^z,y),(y*z)^4]);;
K2 := F / relsK2;;

Order(K2) = 1;

true

Поэтому, если $[x^z,y]=1$ или $[x^{z^{-1}},y]=1$, то $G \simeq K/\langle [x^z,y],yzyzy=zyzyz \rangle \simeq A_7$.

***\% BD: не совсем понятно, почему***

Пусть далее $x^z,x^{z^{-1}} \not \in C_D(y)$. Тогда, как отмечалось выше, $\langle y,x^z \rangle \simeq \langle y,x^{z^{-1}}\rangle \simeq A_5$.
Поэтому $G$ является гомоморфным образом группы
$F=K/\langle (x_1^yx_1)^2,(x_1y)^5,(x_2^yx_2)^2,(x_2y)^5 \rangle$, где $x_1=x^z, x_2=x^{z^{-1}}$. 
При этом,
$F/\langle (xz)^3 \rangle  \simeq A_6$, $F/\langle (xz)^4 \rangle \simeq V$, $F/\langle xzxzx=zxzxz \rangle \simeq U$.

***\% BD: $C_G(y)$, видимо. Почему-то $F/(xz)^3 = 1:$***

In [ ]:
F := FreeGroup("x", "y", "z");;
x := F.1;; y := F.2;; z := F.3;;
x_1 := x^z;; x_2 := x^(z^-1);;

relsK := [x^3,y^3,z^3,comm(x,y),(x^z*x)^2,(x*z)^5];; # продублировал на всякий случай
relsF1 := Concatenation(relsK, [(x_1^y*x_1)^2,(x_1*y)^5,(x_2^y*x_2)^2,(x_2*y)^5]);;

F1 := F / relsF1;;

relsF2 := Concatenation(relsF1, [(x*z)^3]);;
F2 := F / relsF2;;

Size(F2);

A6 := AlternatingGroup(6);;
# iso := IsomorphismGroups(F2, A6);
# ...

1

3. Пусть $x\sim z$. Если $z\sim y$, то $G\simeq Sp_3(3)$. Если $z\sim y^{-1}$, то заменяя $y$ на $y^{-1}$, снова получим $G\simeq Sp_3(3)$.
По доказанному можно считать, что $\langle y,z \rangle \not \in \{3^2,A_4,A_5,SL_2(3) \}$.
Рассмотрим две оставшиеся возможности. Пусть $K=\{ x,y,z \mid x^3,y^3,z^3,[x,y],x\sim z \}$.



In [ ]:
F := FreeGroup("x", "y", "z");;
x := F.1;; y := F.2;; z := F.3;;
relsK := [x^3,y^3,z^3,comm(x,y),sim(x,z)];;

relsK1 := Concatenation(relsK, [(y*z)^3, (y*z^-1)^3]);;
K1 := F / relsK1;;

Size(K1);
Size(Sp(2*3,3));
# IsIsomorphicGroup(K1, Sp(2*3,3)); # Видимо, неверно понимаю

648

9170703360

Заметим, что $K/\langle (yz)^3,(yz^{-1})^3 \rangle \simeq 648$ - уточнить строение этой группы, уточнить является ли она $Sp_3(3)$ и $M$-группой.

Пусть $A=K/\langle y\square z \rangle$ и $B=\langle x,y,y^z \rangle$ --- подгруппа в $A$.

In [1355]:
relsA := Concatenation(relsK, square(y, z));;
A := F / relsA;;

x := A.1;; y := A.2;; z := A.3;;
B := Group(x, y, y^z);;

Перечисляя смежные классы $A$ по $B$ получим подстановочное представление расширения 2-группы
порядка $2^{11}$ посредством $SL_2(9)$, которое не является $M$-группой.


In [ ]:
CosetDecomposition(A, B);

[ [ <identity ...>, y, y^2, y^2*z^-1*y*z*y^2, y*z^-1*y*z*y^2, z^-1*y*z*y^2,       y^2*z^-1*y*z, z^-1*y*z, y*z^-1*y*z, y^2*z^-1*y*z*y, y*z^-1*y*z*y,       z^-1*y*z*y, x, x*y, x*y^2, y^2*z^-1*y*z*x*y^2, y*z^-1*y*z*x*y^2,       z^-1*y*z*x*y^2, y^2*z^-1*y*z*x, z^-1*y*z*x, y*z^-1*y*z*x,       y^2*z^-1*y*z*x*y, y*z^-1*y*z*x*y, z^-1*y*z*x*y, x^2, x^2*y, x^2*y^2,       y^2*z^-1*y*z*x^2*y^2, y*z^-1*y*z*x^2*y^2, z^-1*y*z*x^2*y^2,       y^2*z^-1*y*z*x^2, z^-1*y*z*x^2, y*z^-1*y*z*x^2, y^2*z^-1*y*z*x^2*y,       y*z^-1*y*z*x^2*y, z^-1*y*z*x^2*y, x*z^-1*y*z*y^2, x*y*z^-1*y*z*y^2,       x*y^2*z^-1*y*z*y^2, x*y*z^-1*y*z*x^2*z^-1*y*z*x*y, (x*z^-1*y*z*x)^2*y,       x*y^2*z^-1*y*z*x^2*z^-1*y*z*x*y, y^2*z^-1*y*z*x*z^-1*y*z*y^2,       z^-1*y*z*x*z^-1*y*z*y^2, y*z^-1*y*z*x*z^-1*y*z*y^2,       y^2*z^-1*y*z*x*z^-1*y^2*z*y, y*z^-1*y*z*x*z^-1*y^2*z*y,       z^-1*y*z*x*z^-1*y^2*z*y, x*z^-1*y*z*x^2*z^-1*y*z*y,       x*y^2*z^-1*y*z*x^2*z^-1*y*z*y, x*y*z^-1*y*z*x^2*z^-1*y*z*y,       x*y*z^-1*y*z*x^2*y^2*z^-1*y*z, x*

Положим $w=y^{zxz^{-1}yz^{-1}xyz^{-1}xz}$ и рассмотрим
фактор-группу $C=A/\langle x \sim w \rangle$ и ее подгруппу $D=\langle x,y,y^z \rangle$.
Перечисляя смежные классы $C$ по $D$ получим группу $V$.
Нужно разобраться, является ли получившееся представление точным.

In [ ]:
x := F.1;; y := F.2;; z := F.3;;

w := y^(z*x*z^-1*y*z^-1*x*y*z^-1*x*z);;

relsC := Concatenation(relsA, [sim(x, w)]);;
C := F / relsC;;

x := C.1;; y := C.2;; z := C.3;;
D := Group(x, y, y^z);;

V := CosetDecomposition(C, D);

[ [ <identity ...>, y, y^2, x^2*z^-1*y*z*x, x^2*y*z^-1*y*z*x,       x^2*y^2*z^-1*y*z*x, x^2*y*z^-1*y*z*x*y, x^2*z^-1*y*z*x*y,       x^2*y^2*z^-1*y*z*x*y, x^2*y*z^-1*y*z*x*y^2, x^2*z^-1*y*z*x*y^2,       x^2*y^2*z^-1*y*z*x*y^2, z^-1*y*z*x^2*z^-1*y^2*z*x*z^-1*y*z,       y^2*z^-1*y*z*x^2*z^-1*y^2*z*x*z^-1*y*z,       y*z^-1*y*z*x^2*z^-1*y^2*z*x*z^-1*y*z, y*z^-1*y*z*x^2*y*z^-1*y*z*x,       z^-1*y*z*x^2*y*z^-1*y*z*x, y^2*z^-1*y*z*x^2*y*z^-1*y*z*x,       z^-1*y*z*x^2*z^-1*y^2*z*x, y*z^-1*y*z*x^2*z^-1*y^2*z*x,       y^2*z^-1*y*z*x^2*z^-1*y^2*z*x, z^-1*y*z*x^2*z^-1*y^2*z*x*y,       y*z^-1*y*z*x^2*z^-1*y^2*z*x*y, y^2*z^-1*y*z*x^2*z^-1*y^2*z*x*y,       x*(x*y*z^-1*y*z)^2, x^2*z^-1*y*z*x*y*z^-1*y*z,       x^2*y^2*z^-1*y*z*x*y*z^-1*y*z, z^-1*y*z, y*z^-1*y*z, y^2*z^-1*y*z,       x*(x*z^-1*y*z)^2, x^2*y*z^-1*y*z*x*z^-1*y*z,       x^2*y^2*z^-1*y*z*x*z^-1*y*z, x^2*y*z^-1*y*z*x*y^2*z^-1*y*z,       x^2*z^-1*y*z*x*y^2*z^-1*y*z, x*(x*y^2*z^-1*y*z)^2,       x^2*z^-1*y*z*x*z^-1*y^2*z, x^2*y*z^-1*y*z*x*z^-1*y^

Возможно удобно все вычисления вынести в таблицу и писать доказательство, ссылаясь на нее. $\square$